compose/config/config.py

```diff
@@ -256,39 +258,44 @@ def load(config_details):
     config_details = config_details._replace(config_files=processed_files)
 
     main_file = config_details.config_files[0]
-    volumes = load_volumes(config_details.config_files)
+    volumes = load_mapping(config_details.config_files, 'volumes', 'Volume')
+    networks = load_mapping(config_details.config_files, 'networks', 'Network')
     service_dicts = load_services(
         config_details.working_dir,
         main_file.filename,
         [file.get_service_dicts() for file in config_details.config_files],
         main_file.version)
-    return Config(main_file.version, service_dicts, volumes)
+    return Config(main_file.version, service_dicts, volumes, networks)
 
 
-def load_volumes(config_files):
-    volumes = {}
+def load_mapping(config_files, key, entity_type):
+    mapping = {}
+
     for config_file in config_files:
-        for name, volume_config in config_file.get_volumes().items():
-            volumes[name] = volume_config or {}
-            if not volume_config:
+        for name, config in config_file.config.get(key, {}).items():
```

Comment:
I think there's a subtle edge case here for V1 config. If someone has a service named `volume` or `network` this is going to crash, or break in some unexpected way.

I think we should use the `get_volumes()` function I added to `ConfigFile`, and create a similar one for `get_networks()`. They switch on version, so we won't hit that case.

Instead of `key` you can take `func_name` and do:

`getattr(config_file, func_name)().items()`